<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>BlockExplorerFrame</class>
 <widget class="QFrame" name="BlockExplorerFrame">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>950</width>
    <height>627</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Frame</string>
  </property>
  <property name="frameShape">
   <enum>QFrame::StyledPanel</enum>
  </property>
  <property name="frameShadow">
   <enum>QFrame::Raised</enum>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="spacing">
    <number>0</number>
   </property>
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>12</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QTabWidget" name="m_blockExplorerTabBar">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="m_blockchainTab">
      <attribute name="title">
       <string>Blockchain</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <property name="spacing">
        <number>0</number>
       </property>
       <property name="leftMargin">
        <number>25</number>
       </property>
       <property name="topMargin">
        <number>15</number>
       </property>
       <property name="rightMargin">
        <number>25</number>
       </property>
       <property name="bottomMargin">
        <number>12</number>
       </property>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <property name="spacing">
          <number>10</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <item>
          <spacer name="horizontalSpacer_2">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item alignment="Qt::AlignBottom">
          <widget class="WalletGui::WalletTinyLinkLikeButton" name="m_resetFilterButton">
           <property name="cursor">
            <cursorShape>PointingHandCursor</cursorShape>
           </property>
           <property name="text">
            <string>Reset</string>
           </property>
           <property name="flat">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item alignment="Qt::AlignBottom">
          <widget class="WalletGui::WalletTinyLinkLikeButton" name="m_blockSearchButton">
           <property name="cursor">
            <cursorShape>PointingHandCursor</cursorShape>
           </property>
           <property name="text">
            <string>Show search</string>
           </property>
           <property name="checkable">
            <bool>true</bool>
           </property>
           <property name="flat">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Fixed</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>7</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QFrame" name="m_filterFrame">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
         <property name="frameShape">
          <enum>QFrame::NoFrame</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>5</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>30</number>
          </property>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_7">
            <property name="spacing">
             <number>0</number>
            </property>
            <property name="rightMargin">
             <number>0</number>
            </property>
            <item>
             <widget class="QFrame" name="m_filterPeriodComboFrame">
              <property name="frameShape">
               <enum>QFrame::NoFrame</enum>
              </property>
              <property name="frameShadow">
               <enum>QFrame::Raised</enum>
              </property>
              <layout class="QHBoxLayout" name="m_filter_combo_frame">
               <property name="spacing">
                <number>8</number>
               </property>
               <property name="leftMargin">
                <number>0</number>
               </property>
               <property name="topMargin">
                <number>0</number>
               </property>
               <property name="rightMargin">
                <number>25</number>
               </property>
               <property name="bottomMargin">
                <number>0</number>
               </property>
               <item>
                <widget class="WalletGui::WalletNormalGrayTextLabel" name="m_filterTextLabel">
                 <property name="text">
                  <string>PERIOD</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QComboBox" name="m_filterCombo">
                 <property name="focusPolicy">
                  <enum>Qt::NoFocus</enum>
                 </property>
                 <item>
                  <property name="text">
                   <string>All time</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>Last hour</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>Last day</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>Last week</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>Last month</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>Last year</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>Custom</string>
                  </property>
                 </item>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <widget class="QFrame" name="m_filterPeriodFrame">
              <property name="frameShape">
               <enum>QFrame::NoFrame</enum>
              </property>
              <property name="frameShadow">
               <enum>QFrame::Raised</enum>
              </property>
              <layout class="QHBoxLayout" name="m_filter_period_frame_layout">
               <property name="spacing">
                <number>8</number>
               </property>
               <property name="leftMargin">
                <number>0</number>
               </property>
               <property name="topMargin">
                <number>0</number>
               </property>
               <property name="rightMargin">
                <number>25</number>
               </property>
               <property name="bottomMargin">
                <number>0</number>
               </property>
               <item>
                <widget class="WalletGui::WalletNormalGrayTextLabel" name="m_filterBeginTextLabel">
                 <property name="text">
                  <string>FROM</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QDateTimeEdit" name="m_filterBeginDtedit">
                 <property name="enabled">
                  <bool>true</bool>
                 </property>
                 <property name="locale">
                  <locale language="English" country="UnitedStates"/>
                 </property>
                 <property name="displayFormat">
                  <string>dd MMM yyyy, hh:mm</string>
                 </property>
                 <property name="calendarPopup">
                  <bool>true</bool>
                 </property>
                 <property name="timeSpec">
                  <enum>Qt::UTC</enum>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="WalletGui::WalletNormalGrayTextLabel" name="m_filterEndTextLabel">
                 <property name="text">
                  <string>TO</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QDateTimeEdit" name="m_filterEndDtedit">
                 <property name="enabled">
                  <bool>true</bool>
                 </property>
                 <property name="locale">
                  <locale language="English" country="UnitedStates"/>
                 </property>
                 <property name="displayFormat">
                  <string>dd MMM yyyy, hh:mm</string>
                 </property>
                 <property name="calendarPopup">
                  <bool>true</bool>
                 </property>
                 <property name="timeSpec">
                  <enum>Qt::UTC</enum>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <widget class="QFrame" name="m_filterHashFrame">
              <property name="frameShape">
               <enum>QFrame::NoFrame</enum>
              </property>
              <property name="frameShadow">
               <enum>QFrame::Raised</enum>
              </property>
              <layout class="QHBoxLayout" name="m_filer_hash_frame">
               <property name="spacing">
                <number>8</number>
               </property>
               <property name="leftMargin">
                <number>0</number>
               </property>
               <property name="topMargin">
                <number>0</number>
               </property>
               <property name="rightMargin">
                <number>10</number>
               </property>
               <property name="bottomMargin">
                <number>0</number>
               </property>
               <item>
                <widget class="WalletGui::WalletNormalGrayTextLabel" name="m_filterHashTextLabel">
                 <property name="text">
                  <string>HASH / HEIGHT</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="m_filterHashEdit"/>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <widget class="WalletGui::WalletOkButton" name="m_applyFilterButton">
              <property name="text">
               <string>Apply</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Fixed</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>7</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="WalletGui::WalletTableView" name="m_blockExplorerView">
         <property name="focusPolicy">
          <enum>Qt::NoFocus</enum>
         </property>
         <property name="frameShape">
          <enum>QFrame::NoFrame</enum>
         </property>
         <property name="verticalScrollBarPolicy">
          <enum>Qt::ScrollBarAlwaysOff</enum>
         </property>
         <property name="alternatingRowColors">
          <bool>true</bool>
         </property>
         <property name="selectionMode">
          <enum>QAbstractItemView::NoSelection</enum>
         </property>
         <property name="selectionBehavior">
          <enum>QAbstractItemView::SelectItems</enum>
         </property>
         <property name="textElideMode">
          <enum>Qt::ElideMiddle</enum>
         </property>
         <property name="showGrid">
          <bool>false</bool>
         </property>
         <property name="wordWrap">
          <bool>false</bool>
         </property>
         <property name="cornerButtonEnabled">
          <bool>false</bool>
         </property>
         <attribute name="horizontalHeaderHighlightSections">
          <bool>false</bool>
         </attribute>
         <attribute name="verticalHeaderVisible">
          <bool>false</bool>
         </attribute>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <property name="spacing">
          <number>10</number>
         </property>
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>5</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="WalletGui::WalletNavigationButton" name="m_prevPageButton">
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>30</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="WalletGui::WalletNavigationButton" name="m_nextPageButton">
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
         <item>
          <widget class="WalletGui::WalletNavigationButton" name="m_topButton">
           <property name="text">
            <string>Back to Top</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="m_transactionPoolTab">
      <attribute name="title">
       <string>Transaction pool</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <property name="spacing">
        <number>0</number>
       </property>
       <property name="leftMargin">
        <number>25</number>
       </property>
       <property name="topMargin">
        <number>23</number>
       </property>
       <property name="rightMargin">
        <number>25</number>
       </property>
       <property name="bottomMargin">
        <number>12</number>
       </property>
       <item>
        <widget class="WalletGui::WalletTableView" name="m_transactionPoolView">
         <property name="focusPolicy">
          <enum>Qt::NoFocus</enum>
         </property>
         <property name="frameShape">
          <enum>QFrame::NoFrame</enum>
         </property>
         <property name="selectionMode">
          <enum>QAbstractItemView::NoSelection</enum>
         </property>
         <property name="verticalScrollMode">
          <enum>QAbstractItemView::ScrollPerPixel</enum>
         </property>
         <property name="showGrid">
          <bool>false</bool>
         </property>
         <property name="wordWrap">
          <bool>false</bool>
         </property>
         <property name="cornerButtonEnabled">
          <bool>false</bool>
         </property>
         <attribute name="horizontalHeaderHighlightSections">
          <bool>false</bool>
         </attribute>
         <attribute name="verticalHeaderVisible">
          <bool>false</bool>
         </attribute>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>WalletGui::WalletTinyLinkLikeButton</class>
   <extends>QPushButton</extends>
   <header>Gui/Common/WalletLinkLikeButton.h</header>
  </customwidget>
  <customwidget>
   <class>WalletGui::WalletOkButton</class>
   <extends>QPushButton</extends>
   <header>Gui/Common/WalletOkButton.h</header>
  </customwidget>
  <customwidget>
   <class>WalletGui::WalletNormalGrayTextLabel</class>
   <extends>QLabel</extends>
   <header>Gui/Common/WalletTextLabel.h</header>
  </customwidget>
  <customwidget>
   <class>WalletGui::WalletTableView</class>
   <extends>QTableView</extends>
   <header>Gui/Common/WalletTableView.h</header>
  </customwidget>
  <customwidget>
   <class>WalletGui::WalletNavigationButton</class>
   <extends>QPushButton</extends>
   <header>Gui/Common/WalletNavigationButton.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>m_nextPageButton</sender>
   <signal>clicked()</signal>
   <receiver>BlockExplorerFrame</receiver>
   <slot>nextPage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>726</x>
     <y>591</y>
    </hint>
    <hint type="destinationlabel">
     <x>384</x>
     <y>310</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_prevPageButton</sender>
   <signal>clicked()</signal>
   <receiver>BlockExplorerFrame</receiver>
   <slot>prevPage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>44</x>
     <y>591</y>
    </hint>
    <hint type="destinationlabel">
     <x>384</x>
     <y>310</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_blockSearchButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>BlockExplorerFrame</receiver>
   <slot>showSearch(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>712</x>
     <y>39</y>
    </hint>
    <hint type="destinationlabel">
     <x>384</x>
     <y>310</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_applyFilterButton</sender>
   <signal>clicked()</signal>
   <receiver>BlockExplorerFrame</receiver>
   <slot>applyFilter()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>814</x>
     <y>212</y>
    </hint>
    <hint type="destinationlabel">
     <x>473</x>
     <y>313</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_resetFilterButton</sender>
   <signal>clicked()</signal>
   <receiver>BlockExplorerFrame</receiver>
   <slot>resetFilterClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>894</x>
     <y>212</y>
    </hint>
    <hint type="destinationlabel">
     <x>473</x>
     <y>313</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_filterCombo</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>BlockExplorerFrame</receiver>
   <slot>filterChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>144</x>
     <y>75</y>
    </hint>
    <hint type="destinationlabel">
     <x>473</x>
     <y>313</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_topButton</sender>
   <signal>clicked()</signal>
   <receiver>BlockExplorerFrame</receiver>
   <slot>lastPage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>862</x>
     <y>597</y>
    </hint>
    <hint type="destinationlabel">
     <x>473</x>
     <y>313</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_filterHashEdit</sender>
   <signal>returnPressed()</signal>
   <receiver>m_applyFilterButton</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>813</x>
     <y>77</y>
    </hint>
    <hint type="destinationlabel">
     <x>877</x>
     <y>79</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_blockExplorerView</sender>
   <signal>doubleClicked(QModelIndex)</signal>
   <receiver>BlockExplorerFrame</receiver>
   <slot>showBlockDetails(QModelIndex)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>474</x>
     <y>368</y>
    </hint>
    <hint type="destinationlabel">
     <x>474</x>
     <y>313</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_transactionPoolView</sender>
   <signal>doubleClicked(QModelIndex)</signal>
   <receiver>BlockExplorerFrame</receiver>
   <slot>showTransactionDetails(QModelIndex)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>474</x>
     <y>333</y>
    </hint>
    <hint type="destinationlabel">
     <x>474</x>
     <y>313</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>nextPage()</slot>
  <slot>prevPage()</slot>
  <slot>firstPage()</slot>
  <slot>lastPage()</slot>
  <slot>showBlockDetails(QModelIndex)</slot>
  <slot>showSearch(bool)</slot>
  <slot>applyFilter()</slot>
  <slot>resetFilterClicked()</slot>
  <slot>filterChanged(int)</slot>
  <slot>showTransactionDetails(QModelIndex)</slot>
 </slots>
</ui>
